xcdat/spatial.py

```diff
@@ -729,11 +741,48 @@ def _averager(self, data_var: xr.DataArray, axis: List[SpatialAxis]):
         """
         weights = self._weights.fillna(0)
```

Comment:
I don't think this does anything.

Reply:
I recall we ran into issues with Xarray's weighted mean API when weights contain `np.nan`. The notes above it say:

https://github.com/xCDAT/xcdat/blob/cdae8263b0ce279ffebc36cad98a95e3ee20a5e5/xcdat/spatial.py#L727-L728